### PR TITLE
Issue #572: Improving the podcast trash and podcast duplicate process

### PR DESCRIPTION
**Pull request by zahardev**
I'm not sure about the restoring from trash part, cannot test it on my local site. The problem is that on my dev site, when I restore podcast from the trash, I get a 500 error from the API when trying to recreate the episode (`castos_handler::upload_podcast_to_podmotor`). Can we check why API returns a 500 error there?

### CODE REVIEW

File: php/classes/handlers/class-castos-handler.php [hunk 1]
```diff
@@ -346,7 +346,7 @@ public function upload_podcast_to_podmotor( $post ) {
 
 		$this->logger->log( 'Upload Podcast Response', $response_object );
 
-		if ( ! isset( $response_object->status ) || ! $response_object->status ) {
+		if ( ! isset( $response_object->success ) || ! $response_object->success ) {
```
Comment (zahardev) [inline php/classes/handlers/class-castos-handler.php:349]: Fixed a bug - $response_object->status is integer, and it returns an integer even if error occurs. So we need to check $response_object->success instead.

**Comment by jonathanbossenger** [comment]
@zahardev let me know if you were able to test this properly now, and if you're happy with it, I can review the code.

**Comment by jonathanbossenger** [comment]
@zahardev I've tested this locally. The reason this is failing is due to how, when the post is trashed, both the Episode and the File record in the Castos database is deleted, so our original plan of just deleting the podmotor_episode_id for trashed posts in WordPress will not work.

Let's discuss this with Craig in the development meeting, and we can make a call as to whether we're going to first update the app to allow for this change or go ahead with deleting those meta ids now and work towards improving this process later.

**Comment by jonathanbossenger** [comment]
@zahardev as discussed, for now, we will delete the podmotor_episode_id, podmotor_file_id, and audio_file metadata from the post, when the episode is trashed. This will require the user to reupload the full episode, including the file to Castos if restored.

Note, this should only be for episodes that have a podmotor_episode_id or podmotor_file_id in the first place, ie the user is using Castos. For non Castos posts, we do nothing.

This is not a problem with duplicated episodes, because the user will upload a new episode anyway.

**Comment by jonathanbossenger** [comment]
@zahardev as discussed, for now, we will delete the podmotor_episode_id, podmotor_file_id, and audio_file metadata from the post, when the episode is trashed. This will require the user to reupload the full episode, including the file to Castos if restored.

Note, this should only be for episodes that have a podmotor_episode_id or podmotor_file_id in the first place, ie the user is using Castos. For non Castos posts, we do nothing.

This is not a problem with duplicated episodes, because the user will upload a new episode anyway.

**Comment by zahardev** [comment]
@jonathanbossenger What should happen if deleting from Castos returns an error? Should we remove podmotor_episode_id and podmotor_file_id meta fields in this case?

**Comment by jonathanbossenger** [comment]
@zahardev yes. Once the call to the Castos API happens, the episode is already trashed in WordPress, so it would be better if we still delete the meta data in WordPress. Otherwise we might end up with more sync issues.

**Comment by jonathanbossenger** [comment]
@zahardev can I confirm your last question has been resolved or the code has been changed to match it?

**Comment by zahardev** [comment]
@jonathanbossenger Changed the PR to match it.

**Review by jonathanbossenger — APPROVED** [review]

**Pull request merged**